Darkly/LDClient.h

```diff
@@ -39,7 +40,27 @@
  * @param default   Default value for feature flag
  * @return the feature flag value
  */
-- (BOOL)toggle:(NSString *)featureName default:(BOOL)defaultValue;
+- (BOOL)toggle:(NSString *)featureKey defaultValue:(BOOL)defaultValue;
+/**
+ * Retrieve a feature flag value. If the configuration for this feature
+ * flag is retrieved from the server that value is returned, otherwise
+ * the defaultValue is returned.
+ *
+ * @param featureName   Name of feature flag
+ * @param default   Default value for feature flag
+ * @return the feature flag value
+ */
+- (NSNumber*)numberVariation:(NSString *)featureKey defaultValue:(NSNumber*)defaultValue;
+/**
+ * Retrieve a feature flag value. If the configuration for this feature
+ * flag is retrieved from the server that value is returned, otherwise
+ * the defaultValue is returned.
+ *
+ * @param featureName   Name of feature flag
+ * @param default   Default value for feature flag
+ * @return the feature flag value
+ */
+- (NSString*)stringVariation:(NSString *)featureKey defaultValue:(NSString*)defaultValue;
```

Comment:
If the server is sending pure json strings, then yes, `NSDictionary` and `NSArray` will cover the two types properly. I'll make two separate functions for each of these.